requests/sessions.py

```diff
@@ -90,7 +90,13 @@ class SessionRedirectMixin(object):
     def get_redirect_target(self, resp):
         """Receives a Response. Returns a redirect URI or ``None``"""
         if resp.is_redirect:
-            return resp.headers['location']
+            location = resp.headers['location']
+            # currently the underling http module on py3 decode headers
+            # in latin-1. this causes incorrect handling of UTF8 encoded
+            # location headers. to solve this we re-encode the location
```

Comment:
So I think this comment needs to have a bit more elaboration. Specifically, we need to talk about why we think we're more likely to get UTF-8 encoded location headers than Latin-1: specifically, because empirical evidence suggests that latin-1 is very rarely used with non-ASCII characters in HTTP headers.